route-search/RouteRequest.swift

```
@@ -0,0 +1,152 @@
+        static let httpContentTypeHeader = "content-type"
+        static let httpCustomHeaderValue = "staging"
+        static let httpCustomHeader = "X-AppGlu-Environment"
+        static let jsonRoutesByStopName = "{ \"params\":{ \"stopName\":\"%%%@%%\" }}"
+        static let jsonStopsByRouteId = "{ \"params\":{ \"routeId\":%d }}"
+        static let jsonDeparturesByRouteId = "{ \"params\":{ \"routeId\":%d }}"
+    }
+    
+    // MARK: - Properties
+    private let loginString: String?
+    private let loginData: NSData?
+    private let base64LoginString: String?
+    
+    // MARK: - Initialization
+    init() {
+        
+        loginString = NSString(format: "%@:%@", RouteRequestConstants.user, RouteRequestConstants.password) as String
+        loginData = loginString!.dataUsingEncoding(NSUTF8StringEncoding)!
+        base64LoginString = loginData!.base64EncodedStringWithOptions(NSDataBase64EncodingOptions.Encoding64CharacterLineLength)
+    }
+    
+    // MARK: - Public interfaces to access ArcTouch API
+    public func findRoutesByStopName(routeName: String, handler: ([Route]?)  -> Void) {
+        
+        var routes = [Route]()
+        let json = NSString(format: RouteRequestConstants.jsonRoutesByStopName, routeName) as String
+        
+        apiPostRequest(RouteRequestConstants.findRoutesByStopNameUrl, body: json) { (data, error) -> Void in
+            
+            if(error != nil) {
+                print("Failed to post request findRoutesByStopName \(error?.localizedDescription)")
+                handler(nil)
+            } else {
+                if let rows = data!["rows"] as? [[String: AnyObject]] {
+                    for row in rows {
+                        routes.append(Route(data: row as NSDictionary))
+                    }
+                    handler(routes)
+                    
+                } else {
+                    print("Failed to convert receipt data: \(data)")
+                    handler(nil)
```

You are swallowing the error and the caller of this method will never know if there was any issue with the request or simply no data was found.